packages/x-components/src/components/animations/create-directional-animation-factory.ts

```diff
@@ -33,4 +33,12 @@ export function createDirectionalAnimationFactory(
   };
 }
 
-export type AnimationOrigin = 'top' | 'bottom' | 'left' | 'right';
+export type AnimationOrigin =
+  | 'top'
+  | 'bottom'
+  | 'left'
+  | 'right'
+  | 'top-to-bottom'
+  | 'bottom-to-top'
+  | 'left-to-right'
+  | 'right-to-left';
```

Comment:
Here, We are talking about the origin of the animation, should we rename the type to AnimationDirection? because `top-to-bottom` is not an origin, I see it as a direction

if we do this change, when can rename the values to be more explicit about what they are going to do:

```suggestion
export type AnimationOrigin =
  | 'top-to-top'
  | 'bottom-to-bottom'
  | 'left-to-left'
  | 'right-to-right'
  | 'top-to-bottom'
  | 'bottom-to-top'
  | 'left-to-right'
  | 'right-to-left';
```

I know that its redundant but I think is clear to understand what they are doing.

Reply:
I was thinking also to change the name, but `AnimationDirection` is confusing, because when is `top`, it is moving to the bottom direction on entering, then it can be confusing, and renaming all the current naming to change `top` by `bottom` and `left` to `right` it is a confusing breaking change.

So, to me is better to leave it as it is.  But up to you